nx-X11/programs/Xserver/hw/nxagent/Window.c

```diff
@@ -466,6 +466,18 @@ FIXME: Do all the windows for which nxagentWindowTopLevel(pWin)
     nxagentRedirectWindow(pWin);
   }
 
+  if (!nxagentWindowTopLevel(pWin))
+  {
+    Atom prop = MakeAtom("NX_REAL_WINDOW", strlen("NX_REAL_WINDOW"), True);
+
+    if (ChangeWindowProperty(pWin, prop, XA_WINDOW, 32, PropModeReplace, 1, nxagentWindowPriv(pWin), 1) != Success)
+        fprintf(stderr, "nxagentCreateWindow: Adding NX_REAL_WINDOW failed.\n");
+#ifdef DEBUG
+    else
+        fprintf(stderr, "nxagentCreateWindow: Added NX_REAL_WINDOW for Window ID %ld.\n", nxagentWindowPriv(pWin)->window);
```

Comment:
xwininfo shows the WID in hex, this shows it in decimal. IMHO hex is better, YMMV.